test/overlay.html

```diff
@@ -347,6 +360,35 @@
         });
       });
     });
+
+    describe('frontmost', () => {
+      var dialog;
+      beforeEach(() => {
+        dialog = fixture('front-most');
+      });
+
+      it('should bring to frontmost on mousedown', (done) => {
+        dialog.opened = true;
+        const dialogOverlay = dialog.$.overlay;
+        const datePicker = dialog.$.overlay.querySelector('vaadin-date-picker');
+    
+        const datePickerPosition = datePicker.getBoundingClientRect();
+        const datePickerOverlay = datePicker.$.overlay;
+
+        datePicker.addEventListener('opened-changed', () => {
+          Polymer.RenderStatus.afterNextRender(datePicker, () => {
+            datePicker.dispatchEvent(new CustomEvent('mousedown', {bubbles: true, composed: true}));
+            requestAnimationFrame(() => {
+              const elementFromPoint = document.elementFromPoint(datePickerPosition.right, datePickerPosition.bottom);
+              expect(datePickerOverlay).to.equal(elementFromPoint);
+              done();
+            });
+          });
+        });
+
+        MockInteractions.tap(datePicker);
```

Comment:
For opening the dropdown initially, you can just use `datePicker.opened = true`

Reply:
done

Reply:
The "opened-changed" listener doesn't make sense because it's fired immediately/synchronously when you set the `opened` to `true`: